contracts/hooks/FullRange.sol

```diff
@@ -140,85 +142,61 @@ contract FullRange is BaseHook {
         }
     }
 
-    function modifyPosition(IPoolManager.PoolKey memory key, IPoolManager.ModifyPositionParams memory params)
+    function modifyPosition(PoolKey memory key, IPoolManager.ModifyPositionParams memory params)
         internal
         returns (BalanceDelta delta)
     {
         // msg.sender is the test contract (aka whoever called addLiquidity/removeLiquidity)
 
         delta = abi.decode(poolManager.lock(abi.encode(CallbackData(msg.sender, key, params))), (BalanceDelta));
-
-        uint256 ethBalance = address(this).balance;
-        if (ethBalance > 0) {
-            CurrencyLibrary.NATIVE.transfer(msg.sender, ethBalance);
-        }
     }
 
-    function hookModifyPosition(IPoolManager.PoolKey memory key, IPoolManager.ModifyPositionParams memory params)
+    function hookModifyPosition(PoolKey memory key, IPoolManager.ModifyPositionParams memory params)
         internal
         returns (BalanceDelta delta)
     {
-        // IERC20Minimal(Currency.unwrap(key.currency0)).approve(address(this), type(uint256).max);
-        // IERC20Minimal(Currency.unwrap(key.currency1)).approve(address(this), type(uint256).max);
+        IERC20Minimal(Currency.unwrap(key.currency0)).approve(address(this), type(uint256).max);
+        IERC20Minimal(Currency.unwrap(key.currency1)).approve(address(this), type(uint256).max);
 
         delta = abi.decode(poolManager.lock(abi.encode(CallbackData(address(this), key, params))), (BalanceDelta));
+    }
 
-        uint256 ethBalance = address(this).balance;
-        if (ethBalance > 0) {
-            CurrencyLibrary.NATIVE.transfer(msg.sender, ethBalance);
+    function _addingLiquidity(address sender, PoolKey memory key, BalanceDelta delta) internal {
```

Comment:
I think a better name for these functions would be `_settleDeltas` and `_takeDeltas` respectively.